nomenclature/config.py

```diff
@@ -1,18 +1,69 @@
 from pathlib import Path
-from typing import Dict, Optional
-from pydantic import BaseModel
+from typing import Optional, Dict
+from pydantic import BaseModel, root_validator, validator
 
 import yaml
+from git import Repo
 
 
 class CodeListConfig(BaseModel):
-    repository: Optional[Path]
+    dimension: str
+    repository: Optional[str]
+    repository_definition_path: Optional[str]
+
+    @root_validator()
+    def set_repository_definition_path(cls, v):
+        if (
+            v.get("repository") is not None
+            and v.get("repository_definition_path") is None
+        ):
+            v["repository_definition_path"] = f"definitions/{v['dimension']}"
+        return v
 
 
 class RegionCodeListConfig(CodeListConfig):
     country: Optional[bool]
 
 
+class Repository(BaseModel):
+    url: str
+    hash: Optional[str]
+    release: Optional[str]
+    path: Optional[Path]
```

Comment:
Suggest to rename to clarify that this is the local (cloned) path...
```suggestion
    local_path: Optional[Path]  # defined via the `repository` name in the configuration
```